kotlin-inject-compiler/test/src/main/kotlin/me/tatarka/inject/ProjectCompiler.kt

```diff
@@ -107,5 +107,6 @@ class TestCompilationResult(private val result: CompilationResult) {
     val success: Boolean
         get() = result.exitCode == KotlinCompilation.ExitCode.OK
 
+    @OptIn(ExperimentalCompilerApi::class)
```

Comment:
compile-testing should have a way to filter output in 0.5.0 that can replace our custom code https://github.com/ZacSweers/kotlin-compile-testing/issues/198#issuecomment-1931317732

Reply:
I'm not sure if that would be a better solution here. [I pushed a commit showing the changes](https://github.com/evant/kotlin-inject/pull/350/commits/bb532681c88e88566193b513292da8734899443a), but the issue is that you can't filter after the fact. You have to specify in the `KotlinCompilation` what the logging level should be, which means you need to pass that to `ProjectCompiler.compile`.

Reply:
True, just left a comment about this. Can wait to see if there's any desire to change anything upstream, but otherwise would prefer to get rid of our code even if it's not quite as nice.

Reply:
Ok, made a [pr](https://github.com/ZacSweers/kotlin-compile-testing/pull/260) upstream, can leave it how it is for now.

Reply:
Pushed a commit addressing this